grails-app/domain/org/pih/warehouse/picklist/PicklistItem.groovy

```diff
@@ -68,7 +68,9 @@ class PicklistItem implements Serializable {
     static transients = ['associatedLocation', 'associatedProducts', 'disableRefresh', 'pickable']
 
     String getAssociatedLocation() {
-        return requisitionItem ? requisitionItem?.requisition?.origin?.id : orderItem?.order?.origin?.id
+        return binLocation?.parentLocation?.id ?:
```

Comment:
@jmiranda What in case of default bin location? Is not it then null? I think the issue with refreshing for all locations will be still present for default bins.

Reply:
There's nothing we can do except find a better way altogether. The idea here is that the associated location for the picklist item might be null, but the parent requisition item won't be null (unless we're deleting that too I guess).

If we see the null location case in other places we just have to use the same pattern (disable refresh through parent object, trigger manually at the end of the database transaction).

The one thing I don't like (but don't have the ability to deal with now) is that we're using the disable refresh / trigger manually pattern in seemingly random places. So within a single request we might execute that multiple times manually when once would have been enough.